### PR TITLE
Feat/fix asset link bug

### PR DESCRIPTION
**Pull request by ruderngespra**

### CODE REVIEW

File: lib/tasks/download-assets.js [hunk 1]
```diff
@@ -55,7 +55,13 @@ export default function downloadAssets (options) {
       }
       const locales = Object.keys(asset.fields.file)
       return Promise.mapSeries(locales, (locale) => {
-        const url = asset.fields.file[locale].url || asset.fields.file[locale].upload
+        const url = asset.fields.file[locale].url
+        if (!url) {
+          task.output = `${figures.warning} asset '${getEntityName(asset)}' doesn't contain an url in path asset.fields.file[${locale}].url`
+          warningCount++
```
Comment (phoebeschmidt) [inline lib/tasks/download-assets.js:61]: I would consider this an error, not a warning. (If they are trying to download all their assets and one of them isnt downloadable, that's an error they should be aware of.)

File: test/unit/tasks/download-assets.test.js [hunk 2]
```diff
@@ -10,13 +10,75 @@ const tmpDirectory = resolve(tmpdir(), 'contentful-import-test')
...
-  .times(2)
+  .times(6)
   .reply(200)
 
 nock('https://images.contentful.com')
   .get('/does-not-exist.png')
   .reply(404)
 
+function getAssets ({ existing = 0, nonExisting = 0, missingUrl = 0 } = {}) {
+  const existingUrl = '//images.contentful.com/kq9lln4hyr8s/2MTd2wBirYikEYkIIc0YSw/7aa4c06f3054996e45bb3f13964cb254/rocka-nutrition.png'
```
Comment (phoebeschmidt) [inline test/unit/tasks/download-assets.test.js:21]: we could probably store these strings as global `const`s in this file and reuse them in the nock calls above (line 12 and 16)

File: test/unit/tasks/download-assets.test.js [hunk 3]
```diff
@@ -84,8 +120,47 @@ test('Downloads assets and properly counts failed attempts', () => {
...
+      expect(output.mock.calls).toHaveLength(5)
+    })
+})
+
+test('it doesn\'t use fileStack url as fallback for the file url and throws a warning output', () => {
+  const task = downloadAssets({
+    exportDir: tmpDirectory
+  })
+  const output = jest.fn()
+  const taskProxy = new Proxy({}, {
```
Comment (phoebeschmidt) [inline test/unit/tasks/download-assets.test.js:134]: let's factor out this taskProxy into a `beforeEach`

File: test/unit/tasks/download-assets.test.js [hunk 4]
```diff
@@ -84,8 +120,47 @@ test('Downloads assets and properly counts failed attempts', () => {
...
+    .then(() => {
+      expect(ctx.assetDownloads).toEqual({
+        successCount: 4,
+        warningCount: 2,
+        errorCount: 0
       })
-      expect(output.mock.calls).toHaveLength(4)
+      expect(output.mock.calls).toHaveLength(6)
+
+      const [missingUrlOut1] = output.mock.calls[0]
```
Comment (phoebeschmidt) [inline test/unit/tasks/download-assets.test.js:160]: If I understand correctly, this assumes that the first two calls to `task.output` are the ones for the assets with a missing `url` property. This makes sense but if someone changes the order of the array in the `getAssets` function this test will fail. 

Can we maybe map over `output.mock.calls` using `array.filter` and then check that it finds two matches?

File: test/unit/tasks/download-assets.test.js [hunk 5]
```diff
@@ -84,8 +120,47 @@ test('Downloads assets and properly counts failed attempts', () => {
...
+      const [missingUrlOut2] = output.mock.calls[1]
+
+      expect(missingUrlOut1.endsWith('asset.fields.file[en-US].url')).toBeTruthy()
```
Comment (phoebeschmidt) [inline test/unit/tasks/download-assets.test.js:163]: not truthy, but true :) 
I think `.toBe(true)` or `.toBeTrue()`

Comment (phoebeschmidt) [reply test/unit/tasks/download-assets.test.js:163]: tobetruthy() would still catch our true value, of course, but I think assertions should be as specific as possible

**Review by phoebeschmidt — CHANGES_REQUESTED** [review]
Nice job! A few small comments

**Review by phoebeschmidt — APPROVED** [review]
👍 nice work

**Comment by Vinz93** [comment]
> 👍 nice work

Dankeschön! and thank you for the detailed review 🙂

**Pull request merged**